sonar-batch/src/main/java/org/sonar/batch/phases/AbstractPhaseExecutor.java

```diff
@@ -49,41 +42,29 @@
   private final DefaultModuleFileSystem fs;
   private final QProfileVerifier profileVerifier;
   private final IssueExclusionsLoader issueExclusionsLoader;
-  private final IssuesReports issuesReport;
-  private final DefaultAnalysisMode analysisMode;
-  private final IssueTransition localIssueTracking;
-  private final IssueCallback issueCallback;
-  private final CpdExecutor cpdExecutor;
 
-  public PhaseExecutor(InitializersExecutor initializersExecutor, PostJobsExecutor postJobsExecutor, SensorsExecutor sensorsExecutor,
+  public AbstractPhaseExecutor(InitializersExecutor initializersExecutor, PostJobsExecutor postJobsExecutor, SensorsExecutor sensorsExecutor,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Constructor has 11 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS00107)